### PR TITLE
Docker container stabilization

### PR DESCRIPTION
**Pull request by EdDev**
Fix the docker container and its run to stabilize network availability and NM loading.

### CODE REVIEW

**Comment by EdDev** [comment]
ci test please

File: automation/run-integration-tests.sh [hunk 1]
```diff
@@ -33,6 +33,8 @@ docker network create $NET0 || true
 docker network create $NET1 || true
 docker network connect $NET0 $CONTAINER_ID
 docker network connect $NET1 $CONTAINER_ID
+docker exec $USE_TTY -i $CONTAINER_ID /bin/bash -c 'nmcli dev; ip addr'
+docker exec $USE_TTY -i $CONTAINER_ID /bin/bash -c 'ip addr flush eth1 && ip addr flush eth2; ping -c 3 8.8.8.8'
```
Comment (tyll) [inline automation/run-integration-tests.sh]: IMHO `ping -c 1` would be enough but pinging the Google DNS also feels wrong.
Why do you use `&&` to chain the two flushes but ignore the second returncode before pinging Google?

Comment (tyll) [reply automation/run-integration-tests.sh]: The NM debug info should include `nmcli con`.

File: automation/run-integration-tests.sh [hunk 2]
```diff
@@ -28,13 +28,14 @@ cd "$EXEC_PATH"
...
 docker network create $NET0 || true
 docker network create $NET1 || true
 docker network connect $NET0 $CONTAINER_ID
 docker network connect $NET1 $CONTAINER_ID
+docker exec $USE_TTY -i $CONTAINER_ID /bin/bash -c 'nmcli dev; ip addr'
+docker exec $USE_TTY -i $CONTAINER_ID /bin/bash -c 'ip addr flush eth1 && ip addr flush eth2; ping -c 3 8.8.8.8'
 
 pyclean
-docker exec $USE_TTY -i $CONTAINER_ID /bin/bash -c 'systemctl start NetworkManager'
+docker exec $USE_TTY -i $CONTAINER_ID /bin/bash -c 'nmcli dev; ip addr'
```
Comment (tyll) [inline automation/run-integration-tests.sh]: The NM debug info should include `nmcli con`.

**Comment by tyll** [comment]
When I run this on Fedora 29 beta, nmcli con reports:

```
DEVICE  TYPE      STATE                                  CONNECTION         
eth2    ethernet  connected                              eth2               
eth1    ethernet  connecting (getting IP configuration)  Wired connection 1 
eth0    ethernet  unmanaged                              --                 
lo      loopback  unmanaged                              -- 
```

**Comment by tyll** [comment]
Can you maybe describe the expected state of the docker container before it will be running the integration tests regarding interfaces and output of nmcli con/nmcli dev/ip addr?

File: automation/run-integration-tests.sh (hunk 1 above)
Comment (EdDev) [reply automation/run-integration-tests.sh]: I split the debug from all the rest.
pinging only 1 packet is enough to check if there is a connectivity or not, but I wanted to give it a chance to recover, trying to retry.
Added the `nmcli con`, thanks.

File: automation/run-integration-tests.sh [hunk 3]
```diff
@@ -33,6 +33,7 @@ docker network create $NET0 || true
 docker network create $NET1 || true
 docker network connect $NET0 $CONTAINER_ID
 docker network connect $NET1 $CONTAINER_ID
+docker exec $USE_TTY -i $CONTAINER_ID /bin/bash -c 'ip addr flush eth1 && ip addr flush eth2'
```
Comment (tyll) [inline automation/run-integration-tests.sh]: There is a typo in the commit message, it says `ethi` instead of `eth`.

**Comment by EdDev** [comment]
ci test please

**Comment by cathay4t** [comment]
*From Nmstate Github Robot*

Test  has

**Pull request merged**